googlemock/CMakeLists.txt.install

```diff
@@ -15,7 +15,17 @@ include_directories(
 add_library(gmock STATIC
   ${gtest_vendor_BASE_DIR}/src/gtest-all.cc
   src/gmock-all.cc)
-target_link_libraries(gmock ${CMAKE_THREAD_LIBS_INIT})
+# When building with asan (i.e. using colcon build --mixin asan-gcc),
+# asan itself provides a "fake" pthread that tricks the pthread
+# detection logic into thinking no link libraries are necessary
+# (see https://wiki.gentoo.org/wiki/AddressSanitizer/Problems#pthread_linking_issues
+# for some additional information).  To work around that, we unconditionally
+# add the -pthread flag for Linux machines so it will always work
+if(UNIX AND NOT APPLE)
+  target_link_libraries(gmock ${CMAKE_THREAD_LIBS_INIT} -pthread)
+else()
+  target_link_libraries(gmock ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
Please keep the equal part outside of the `if` block to avoid unnecessary duplication.

Same below.

Reply:
Done in 3b4118cbad19fab489c8143af373c85526135352

Reply:
Just keep it simple. No need to create a list. Simply invoke the function once unconditionally and once conditionally for `pthread`.

Reply:
Oh, I see.  I didn't realize that calls to `target_link_libraries` were cumulative, not a replacement.  Fixed in bdcff3c